packages/sfpowerscripts-cli/src/commands/profile/retrieve.ts

```diff
@@ -65,7 +65,7 @@ export default class Retrieve extends SfpowerscriptsCommand {
             folders.push(...argFolder);
         }
 
-        Sfpowerkit.initCache();
+        Sfpowerkit.initCache(true);
```

Comment:
@azlam-abdulsalam 
Why did you only optionally reseted the cache for the other two commands and forced it for the retrieve? Why not uniformly? The error can occur with every command. And is there anything against not flushing the cache when changing the org? Or to build the cache key with alias?

Reply:
same question here

Reply:
Reconcile is used in sfpowerscripts using a dual pass approach, so if we reset the second one will be expensive. Hence made optional, it makes sense though while you retrieve to ensure you ignore any existing cache.

 Ideally the fix would be to be have a proper data structure, that caches for every org, as you proposed, but I do think this is a  cheaper option for now to get this bug resolved, and then proceed to fix it later

Reply:
For me thats fine fore a temp fix. How do you decide if a cache is valid or not anymore?